### PR TITLE
Expose `fuzzy_transpositions` parameter in fuzzy queries

### LINKED ISSUE #18348: Expose `fuzzy_transpositions` in fuzzy queries
The `fuzzy_transpositions` parameter is supported by the `match` query (`true` by default) but not by the `multi_match`, `query_string`, `simple_query_string`, or `fuzzy` queries, nor by the completion suggester.

We should add this parameter to all of the above.

### PR DESCRIPTION
**Pull request by alexshadow007**
Add `fuzzy_transpositions` parameter to `multi_match` and `query_string` queries.
Add `fuzzy_transpositions`, `fuzzy_prefix_length` and` fuzzy_max_expansions`
parameters to `simple_query_string` query.
Add `transpositions` parameter to doc for` fuzzy` query.

Closes #18348

### CODE REVIEW

File: docs/reference/query-dsl/query-string-query.asciidoc
Comment (jimczi) [inline docs/reference/query-dsl/query-string-query.asciidoc]: I'd say "set `false` to disable fuzzy transpositions" because the default is true ?

File: docs/reference/query-dsl/simple-query-string-query.asciidoc
Comment (jimczi) [inline docs/reference/query-dsl/simple-query-string-query.asciidoc]: Same here

**Review by jimczi — APPROVED** [review]
Thanks @alexshadow007 !
I left a minor comment but the pr looks good.
I'll let the CI run on this and will merge if it doesn't complain. I also think that it can be merged to 6.x since it's just a new option.

**Comment by jimczi** [comment]
@elasticmachine ok to test

**Comment by alexshadow007** [comment]
@jimczi Updated docs

**Comment by alexshadow007** [comment]
@jimczi I also found that [`fuzzy`](https://github.com/elastic/elasticsearch/blob/927111c91dac5e3cef868eead3cfe8fd4075bee5/core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java#L58) query is using `false` as default value for transpositions. I can change it in another PR, what do you think about it?

**Comment by jimczi** [comment]
Thanks @alexshadow007 ! I'll backport to 6.x shortly.
You can open another PR for the default value in the fuzzy query. I agree that it should be consistent but since it's a breaking change it should target 7.0 only.

**Pull request merged**
